src/Components/dashboard/route/UpdateRouteButton.jsx

```diff
@@ -0,0 +1,242 @@
+
+import Buttons from "../operatorsanddriver/Button.jsx";
+import "./UpdateRoute.css";
+import ListItem from "@mui/material/ListItem";
+import { useDispatch, useSelector } from "react-redux";
+import { updateRoute } from "../../../redux/Action/routes.js";
+import { Dialog } from "@material-ui/core";
+import {ToastContainer, toast} from 'react-toastify';
+
+const un = "Bus stop one, You can add others by hitting + button"
+let id = location.href.split("?id=")[1];
+let routeUuid = decodeURIComponent(id);
+
+const UpdateRouteButton = (props) => {
+
+
+    const notify = (toastMsg) => toast(toastMsg)
+    const [routeCode, setRouteID] = useState("");
+    const [startLocation, setStartPoint] = useState("");
+    const [endLocation, setEndPoint] = useState("");
+    const [stop, setStop] = useState("");
+
+    const routeDetails = useSelector((state) => state.routeDetails);
+    const {loading, route, error } = routeDetails; 
+   
+
+    const [routeIDError, setRouteIDError] = useState(false);
+    const [startPointError, setStartPointError] = useState(false);
+    const [endPointError, setEndPointError] = useState(false);
+    const [stopError, setStopError] = useState(false);
+
+    const [openUpdater, setOpenUpdater] = useState(true);
+    const [open, setOpen] = useState(false);
+    const [updated, setupdated] = useState(false);
+
+    const handleClear = () => {
+        console.log("Content cleared");
+        setRouteID('');
+        setStartPoint('');
+        setEndPoint('');
+    };
+
+    const handleClose = () => {
+        setOpen(false);
+        window.location.reload()
+      };
+    
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+
+ const handleCloseUpdater = () => {
+        setOpenUpdater(false);
+        setOpen(false);
+        
+      };
+
+     
+    const handleAddRoute = () => {       
+
+        const List = document.querySelector(".list-route");
+        const ListItem = document.createElement("li");
+        ListItem.style.border = "1px solid #c4c4c4";
+        ListItem.style.width = "70%";
+        ListItem.style.height = "40px";
+        ListItem.style.paddingLeft = "20px";
+
+        ListItem.innerHTML = stop;
+        List.appendChild(ListItem);
+    };
+
+    
+    const dispatch = useDispatch();
+
+    const handleUpdate = (e) => {
+        e.preventDefault();
+    
+        setRouteIDError(false);
+        setStartPointError(false);
+        setEndPointError(false);
+        setStopError(false);
+        setOpen(false)
+    
+    
+        if (routeCode === "") {
+          setRouteIDError(true)
+        }
+        if (startLocation === "") {
+          setStartPointError(true)
+        }
+        if (startLocation === "") {
+          setEndPointError(true)
+        }
+        if (stop === "") {
+          setStopError(true)
+        }
+        else {                     
+          console.log(routeCode, startLocation, endLocation, stop, name, "yessss");
```

Comment:
Looks great, but let's remove this console.log line.

Reply:
Look like I missed this ones, I have resolved them